### ISSUE TITLE
chrome lighthouse text compression

### ISSUE THREAD

**Issue by pYr0x** [expressjs/compression#142]
i have searched the closed issues and found some others that have the same problem. but no solution
i enabled compression with 
`app.use(compression({ threshold: 0 }));`

* chrome networktab says: no compression, no content-encoding.
```
HTTP/1.1 200 OK
X-Powered-By: Express
Accept-Ranges: bytes
Cache-Control: public, max-age=0
Last-Modified: Sun, 10 Jun 2018 00:30:44 GMT
ETag: W/"1a526-163e71a33c0"
Content-Type: text/css; charset=UTF-8
Vary: Accept-Encoding
Date: Sun, 10 Jun 2018 01:14:59 GMT
Connection: keep-alive
Transfer-Encoding: chunked
```
* chrome lighthouse extension says: Enable Text Compression!
* firefox networktab: no content-encoding on the response

* i turned debugging on `set DEBUG=compression` 
i see on every request: 
```
compression gzip compression +1m
GET /dist/bundles/medicalpad/index.css 200 1.590 ms - -
```

* if i used `curl -i --compressed http://localhost:3000/dist/bundles/medicalpad/index.css`
i see 
```
TP/1.1 200 OK
X-Powered-By: Express
Accept-Ranges: bytes
Cache-Control: public, max-age=0
Last-Modified: Sun, 10 Jun 2018 00:30:44 GMT
ETag: W/"1a526-163e71a33c0"
Content-Type: text/css; charset=UTF-8
Vary: Accept-Encoding
Content-Encoding: gzip    <<<-------
Date: Sun, 10 Jun 2018 01:13:40 GMT
Connection: keep-alive
Transfer-Encoding: chunked
```

so i dont know whats going on there.

i am using:
* express 4.16.0
* compression: 1.7.2
* node: 8.11.2

**Comment by pYr0x** [comment]
i have made a demo https://github.com/pYr0x/express-gzip
even on that simple example, chrome and other browers dont show the Content-Encoding.

if i visit other global website e.g. https://www.nytimes.com/ i see a `content-encoding: gzip` so a bug in chrome can be excluded.

the only difference on the respone headers i found is that:
* content-length is missing
* and there is a **Transfer-Encoding: chunked** header. 

can  `Transfer-Encoding: chunked` be the reason why the Content-Encoding is not shown or ignored by chrome?

@dougwilson is the demo shown you the Content-Encoding? i am on windows 10 with chrome 67.0.3396.79

**Comment by dougwilson** [comment]
Thanks for providing a repo! I an getting `Content-Encoding: gzip` in the Chrome Console just as I would expect from your project:
![image](https://user-images.githubusercontent.com/67512/41201939-75aa177e-6c8f-11e8-877a-b4731325c668.png)

**Comment by pYr0x** [comment]
you are on windows right?
what node and npm version do you have installed?

**Comment by dougwilson** [comment]
Yes, I am on Windows 10. I used Node.js 8.11.2 (installed fresh just for this) so we would be using the exact same Node.js to try and keep every as similar as possible from what you said so far.

**Comment by pYr0x** [comment]
So I have no glue what’s going on... :(

**Comment by dougwilson** [comment]
Same. Variations of this issue have been report many times but either it suddenly started working and the reported doesn't know why or I had to close due to no further progress. If I could reproduce it I could try to track it down, otherwise since you can reproduce it we're all awaiting to hear what the issue is that you're having and where the bug is located.

**Comment by pYr0x** [comment]
ok one last try:
- can you disable all chrome extensions and try it with a clean chrome browser?
- can you tell me the file size that chrome gets with gizp? i will compare that with the filesize i will get.

**Comment by dougwilson** [comment]
Hi @pYr0x sorry I didn't get back to you earlier. I just tried out your additional suggestions and everything is still working fine for me.

I did the following:

1. Installed a brand new Windows 10 on an old machine.
2. Installed a new copy of Google Chrome
3. Installed a fresh copy of Node.js
4. Ran the code I posted above and opened Chrome

This means that nothing I had on my machine before was preset. There were no extensions in Chrome now, as it's a fresh install on a fresh Windows 10 install.

The file size that Chrome is showing with gzip is 87.9 kb.
